spt/features/visualizations/player_trace/tr_config.hpp

```diff
@@ -0,0 +1,173 @@
+#pragma once
+
+#include "spt/features/visualizations/renderer/mesh_renderer.hpp"
+
+#ifdef SPT_MESH_RENDERING_ENABLED
+#define SPT_PLAYER_TRACE_ENABLED
+
+namespace player_trace
+{
+	enum TrSegmentReason : int
+	{
+		TR_SR_FCPS,
+		TR_SR_PLAYER_PORTALLED,
+		TR_SR_SAVELOAD,
+		TR_SR_MAP_TRANSITION,
+		TR_SR_IMPLICIT, // never recorded, only used for drawing
+
+		TR_SR_COLORED_COUNT,
+
+		TR_SR_TRACE_START,
+
+		TR_SR_NONE,
+	};
+
+	enum TrPlayerCameraDrawType
+	{
+		TR_PCDT_FRUSTUM,
+		TR_PCDT_BOX_AND_LINE,
+
+		TR_PCDT_COUNT,
+	};
+
+	struct
```

Comment:
Any particular reason why this file uses C-style structs?

Reply:
I'm not sure I understand the question.
- everything in that file is effectively a global, so it's all gotta be public (so I used structs instead of classes)
- the reason I used globals at all instead of hardcoding all the values is to smooth the making of the imgui interface for the player trace in the future (this way I can just edit those values directly from some separate file and it'll immediately effect how the player trace is drawn)
- the reason it's all in structs instead of being naked globals is because intellisense will make it a bit easier to use something like `trStyles.entities.drawObbCenter` or `trStyles.pathStyle.cones.tickInterval` than naked globals

Reply:
I mean the name being at the end of the struct and not at the beginning?

Reply:
Is that really C-style structs? Those are just named instances of anonymous structs. Googles ai overview gives a good example:
```#include <stdio.h>

struct Employee {
    char name[50];
    struct { // Anonymous struct for address
        char street[50];
        char city[50];
        char state[3];
        int zip_code;
    } address;
};

int main() {
    struct Employee emp = {
        "John Doe",
        {
            "123 Main St",
            "Anytown",
            "CA",
            91234
        }
    };

    printf("Name: %s\n", emp.name);
    printf("Address: %s, %s, %s %d\n", 
           emp.address.street, emp.address.city, emp.address.state, emp.address.zip_code);

    return 0;
}
```